### PR TITLE
Remove unnecessary implementations from MockHandler

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #125423
* #126245
* #126553
* __->__ #126511

Dead implementations are confusing and can cause bugs when people
accidentally hit them.  Better for it to be missing.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/ops_handler.py [hunk 1]
```diff
@@ -748,9 +747,27 @@ def inner(*args):
 
             return inner
 
-        for name, format_string in itertools.chain(
-            magic_methods.items(), inplace_methods.items()
-        ):
+        for name, format_string in {
```
Comment (Skylion007) [inline torch/_inductor/ops_handler.py:750]: Why this chain?

**Review by peterbell10 — APPROVED** [review]

File: torch/_inductor/ops_handler.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/ops_handler.py:750]: err, chain?

**Review by lezcano — APPROVED** [review]
Yay, div is gone!!

**Pull request closed without merging**